sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1789 @@
+would permit the miner to include transactions in blocks that it cannot include today.
+
+Finally, a year of active development of smart contracts on the Stacks
+blockchain highlighted areas for improved support in the Clarity smart
+contracting language. These areas include increasing the visibility of
+burnchain operations (on the Stacks chain, these are *Bitcoin
+operations*) and general improvements to the Clarity programming
+language.
+
+## How to Read this SIP
+
+This SIP specification is structured like a changelog.  Proposed changes to each of these
+areas of the system are described in order of new features, changed features,
+and fixed features.  A rationale is provided for each proposed modification to
+Stacks.
+
+# Specification
+
+Applying these upgrades to the Stacks blockchain requires a
+consensus-breaking network upgrade, in this case, a hard fork. Like
+other such changes, this will require a new Stacks epoch. In this SIP,
+we will refer to this new epoch as Stacks 2.1.
+
+At the onset of Stacks 2.1, the Clarity VM will begin to support
+"Clarity Version 2". This version will include support for the new
+native methods proposed in this SIP (and therefore include new
+*keywords* which cannot be used for method names or variable names).
+New contracts launched in Stacks 2.1 will _default_ to Clarity 2, but
+contract authors will be able to use a new contract-publish transaction type to
+indicate if a contract should specifically launch using
+Clarity 1 or Clarity 2.
+
+Additionally, a new `pox-2` contract will be
+published by the boot address. The `stacks-node` will use the new
+`pox-2` contract for determining PoX reward sets and governing PoX
+locks.  Similarly, new rules for block validation take effect at the onset of Stacks
+2.1.
+
+## Stacking
+
+This section of the SIP specifies the new and changed behaviors of the
+PoX-2 contract and also provides new use cases that demonstrate how
+the new PoX-2 contract could be used.
+
+### Overview
+
+The new PoX contract operates exclusively with PoX state that was
+created in the new contract. It will not "import" state from the
+original PoX contract. In order to allow for this, a particular reward
+cycle `N` is chosen as the "Last PoX-1" reward cycle, and then the
+"First PoX-2" reward cycle is the subsequent cycle (`N+1`).
+
+This defines three periods of PoX operation:
+
+Period 1 | Period 2 | Period 3
+-- | -- | --
+2.0 Consensus Rules in Effect | 2.1 Consensus Rules enacted, but first PoX-2 reward cycle has not begun | First PoX-2 reward cycle has begun
+_This is the period of time before the 2.1 fork._ | _This is after the 2.1 fork in cycle N, but before cycle (N+1)._ | _This is the start of cycle (N+1), and all cycles afterward.  The original PoX contract state will no longer have any impact on reward sets, account lock status, etc._
+
+- Every account that is locked by the original contract for cycle `N`
+  and beyond is unlocked at the end of Cycle `N`.
```

Comment:
Should amend this to say:
"Every account that is locked by the original contract for cycle `N` and beyond is unlocked at the end of Cycle `N`, unless a call to `pox-2.stack-extend` (or `pox-2.delegate-stack-extend`) was made before the end of cycle N."

Reply:
Does this mean we can "extend" (via PoX-2) stacked funds, which were stacked by PoX-1?

Reply:
I don't think so. The original language hinting towards that was removed recently with Jude's latest pass at clarifying this section. @jcnelson can you confirm?